### DISCUSSION TITLE
SMTP Errors when sending Emails from Flows/Invites

### DISCUSSION THREAD

**Discussion by salohcin714** [directus/directus#17243] [Archive]
Directus version: 9.16.1
Node version: 16.14.0

Receiving the following error when trying to send an invite email:

{
  "message": "140685257054144:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:\n",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}

I am also having issues sending emails through flows so I believe the email is just not working correctly.

Here are my current settings

```
# What to use to send emails. One of
# sendmail, smtp, mailgun, ses.
EMAIL_TRANSPORT="smtp"
EMAIL_SENDMAIL_NEW_LINE="unix"
EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"

## Email (Sendmail Transport)

# What new line style to use in sendmail ["unix"]
EMAIL_SENDMAIL_NEW_LINE="unix"

# Path to your sendmail executable ["/usr/sbin/sendmail"]
EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"

## Email (SMTP Transport)
EMAIL_SMTP_HOST="smtp.office365.com"

# Use SMTP pooling
EMAIL_SMTP_POOL=true
EMAIL_SMTP_PORT=587
EMAIL_SMTP_SECURE=false # Use TLS
EMAIL_SMTP_IGNORE_TLS=false
EMAIL_SMTP_USER="username" <= These are actually filled. with the correct information
EMAIL_SMTP_PASSWORD="password"
```

I tested using nodemailer in an outside script to make sure it wasn't an issue with my version of node (16.14.0), and the email sends perfectly fine with the same information.

**Comment by paescuj** [comment]
See https://github.com/directus/directus/discussions/11236#discussioncomment-2034584

**Reply by salohcin714** [reply]
Thanks, but I already have EMAIL_SMTP_SECURE set to false. Nodemailer has no problem when ran outside of Directus.

**Reply by paescuj** [reply]
Ah, right! So this error message means there's a misconfiguration as for the TLS settings. Are these the exact same settings as you've used in nodemailer?

Maybe this helps: https://github.com/nodemailer/nodemailer/issues/1219

**Reply by salohcin714** [reply]
The settings are exactly the same. I thought maybe it was an issue with openssl on the server, but running the same nodemailer script there also succeeded

**Reply by paescuj** [reply]
Well, I guess there must be a difference somewhere. You might want to double check the settings and ensure that those aren't getting overwritten anywhere (e.g. at another place in the `.env` file, or by an environment variable). Also make sure the runtime context is the same as with the nodemailer test you've made (perhaps you're using `nvm` or or something like this).

**Reply by salohcin714** [reply]
I see no difference 


Email section of .env
```
####################################################################################################
### Email

# Email address from which emails are sent ["no-reply@directus.io"]
EMAIL_FROM="********"

# What to use to send emails. One of
# sendmail, smtp, mailgun, ses.
EMAIL_TRANSPORT="smtp"
#EMAIL_SENDMAIL_NEW_LINE="unix"
#EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"

## Email (Sendmail Transport)

# What new line style to use in sendmail ["unix"]
#EMAIL_SENDMAIL_NEW_LINE="unix"

# Path to your sendmail executable ["/usr/sbin/sendmail"]
#EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"

## Email (SMTP Transport)
EMAIL_SMTP_HOST="smtp.office365.com"

# Use SMTP pooling
EMAIL_SMTP_POOL=true
EMAIL_SMTP_PORT=587
EMAIL_SMTP_SECURE=false # Use TLS
EMAIL_SMTP_IGNORE_TLS=false
EMAIL_SMTP_USER="********"
EMAIL_SMTP_PASSWORD="********"

## Email (Mailgun Transport)
# EMAIL_MAILGUN_API_KEY="key-1234123412341234"
# EMAIL_MAILGUN_DOMAIN="a domain name from https://app.mailgun.com/app/sending/domains"
```

Configuration I am using to test the nodemailer script

```
const transporter = nodemailer.createTransport({
  host: 'smtp.office365.com',
  port: 587,
  pool: true,
  secure: false,
  ignoreTLS: false,
  auth: {
    user: '********',
    pass: '********'
  }
})
```


I am using pm2 to serve directus, but there is only one version of node installed on the system

**Reply by paescuj** [reply]
Okay, looks good so far. Have you checked this as well?:
> [...] You might want to double check the settings and ensure that those aren't getting overwritten anywhere (e.g. at another place in the .env file, or by an environment variable). [...]

**Reply by salohcin714** [reply]
I checked and there are not any other Email environment variables present outside of that section (that is also the end of the file). I also checked my PM2 setup and I am just running `npx directus start` without and environment variables

**Reply by paescuj** [reply]
Hmm, that's really strange. As already mentioned, I'd expect it to work the same way, as long as the context and config are the same as well. These are the only ideas I've left:
- Latest version of Directus uses nodemailer 6.8.0. Although the changelog has no entries in this regard, it could work differently because you're using a different (newer) version? Note that nodemailer in Directus is updated to 6.9.1 with the next release (https://github.com/directus/directus/pull/17353).
- Very unlikely, but maybe it has something to do with it `EMAIL_FROM` (missing in your nodemailer config) or you could try to disable `EMAIL_SMTP_POOL` (because Directus is running longer while your nodemailer call probably exits after one email).

**Comment by notflip** [comment]
@salohcin714 Where can you see the details of this error? I'm also receiving errors when inviting using email, but I only see a short message on the screen saying `[INTERNAL_SERVER_ERROR] write EPIPE`

**Reply by davay42** [reply]
Got same error with my new directus 10.8 after moving from previous server. Used previous SMTP settings, that worked just fine with my prev 10.7 server....

**Reply by davay42** [reply]
Getting this error is a simple send Email flow 
`{
  "name": "Error",
  "message": "write EPIPE"
}`

**Comment by singatias** [comment]
same here running docker:
```
[11:25:38.632] ERROR: Could not send password reset mail
    err: {
      "type": "Error",
      "message": "write EPIPE",
      "stack":
          Error: write EPIPE
              at afterWriteDispatched (node:internal/stream_base_commons:160:15)
              at writeGeneric (node:internal/stream_base_commons:151:3)
              at Socket._writeGeneric (node:net:962:11)
              at Socket._write (node:net:974:8)
              at writeOrBuffer (node:internal/streams/writable:392:12)
              at _write (node:internal/streams/writable:333:10)
              at Writable.write (node:internal/streams/writable:337:10)
              at LastNewline.ondata (node:internal/streams/readable:809:22)
              at LastNewline.emit (node:events:517:28)
              at addChunk (node:internal/streams/readable:368:12)
      "errno": -32,
      "code": "EPIPE",
      "syscall": "write"
    }

**Reply by tocomo** [reply]
I am getting the same error. I tried different mail providers with the same result. How can we improve logging to get to error cause?

**Reply by chrisschuller** [reply]
Any updates or solutions on this one?

**Comment by ra7bi** [comment]
Same here !

**Reply by zafon18** [reply]
Same here !

**Comment by vratskyi** [comment]
Same problem, even after full smtp configuration via docker compose I get sending error, is it possible to disable email sending in principle ? I have my own microservice with templates also built on nodemailer running in docker and fully working, network between containers, how to disable email sending in directus for example if I access /users/invite and want to get invite_url, but directus doesn't give link until email is sent.

By the way, my email sending microservice works and is configured on google smtp with 465 port, but I get a sending error when I configure directus the same way.

Here are all the settings I used last time but with port 587:

```yml
 EMAIL_FROM: ${EMAIL_USER}
 EMAIL_TRANSPORT: "smtp"
 EMAIL_SMTP_HOST: "smtp.gmail. com"
 EMAIL_SMTP_PORT: "587"
 EMAIL_SMTP_USER: ${EMAIL_USER}
 EMAIL_SMTP_PASSWORD: ${EMAIL_PASSWORD}
 EMAIL_SMTP_POOL: false
 EMAIL_SMTP_SECURE: false
 EMAIL_VERIFY_SETUP: true
 EMAIL_SMTP_IGNORE_TLS: false
 EMAIL_SMTP_TLS: true
 EMAIL_SMTP_NAME: "Some dude"
```

**Comment by ra7bi** [comment]
This is working with me 
```
      # MailDev
      EMAIL_TRANSPORT: "smtp"
      EMAIL_SMTP_HOST: "maildev"
      EMAIL_SMTP_PORT: 1025
      EMAIL_SMTP_PASSWORD: ""
      EMAIL_SMTP_IGNORE_TLS: true
      EMAIL_SMTP_NAME: "Directus"
      EMAIL_SMTP_USER: "no-reply@example.com"
```
